src/troika/sites/sge.py

```diff
@@ -0,0 +1,266 @@
+import re
+import signal
+import time
+
+from .. import InvocationError, RunError
+from ..connection import PIPE
+from ..parser import BaseParser, ParseError
+from ..utils import check_retcode, command_as_list
+from .base import Site
+
+_logger = logging.getLogger(__name__)
+
+
+def _split_sge_directive(arg):
+    """Split the argument of a SGE directive
+
+    >>> _split_sge_directive(b"-o foo")
+    (b'-o', b'foo')
+    >>> _split_sge_directive(b"-N job")
+    (b'-N', b'job')
+    >>> _split_sge_directive(b"-V")
+    (b'-V', None)
+    """
+    m = re.match(rb"(\S+)(\s+)?(.*)?$", arg)
+    if m is None:
+        raise RunError(r"Malformed qsub argument: {arg!r}")
+    key, sep, val = m.groups()
+    if sep is None:
+        assert val == b""
+        val = None
+    return key, val
+
+
+class SGEDirectiveParser(BaseParser):
+    """Parser that processes a script to extract SGE directives
+
+    Parameters
+    ----------
+    drop_keys: Iterable[bytes]
+        Directives to ignore, e.g. ``[b'-o', b'-e']``
+
+    Members
+    -------
+    data: collections.OrderedDict[bytes, (bytes or None, bytes)]
+        Directives that have been parsed. The first item of the dict value is
+        the parsed value, if any, and the second is the full line, including
+        the line terminator.
+    """
+
+    DIRECTIVE_RE = re.compile(rb"^#\s*$\s+(.+)$")
+
+    def __init__(self, drop_keys=None):
+        super().__init__()
+        self.data = OrderedDict()
+        if drop_keys is None:
+            drop_keys = []
+        self.drop_keys = set(drop_keys)
+
+    def feed(self, line):
+        """Process the given line
+
+        See ``BaseParser.feed``
+        """
+        m = self.DIRECTIVE_RE.match(line)
+        if m is None:
+            return False
+
+        key, value = _split_sge_directive(m.group(1))
+        if key not in self.drop_keys:
+            self.data[key] = (value, line)
+
+        return True
+
+
+def _translate_export_vars(value):
+    if value == b"all":
+        return b"-V"
+    if value == b"none":
+        return
+    return b"-v %s" % value
+
+
+def _translate_mail_type(value):
+    trans = {b"none": b"n", b"begin": b"b", b"end": b"e", b"fail": b"a"}
+    vals = value.split(b",")
+    newvals = []
+    for val in vals:
+        newval = trans.get(val.lower())
+        if newval is None:
+            _logger.warn("Unknown mail_type value %r", val)
+            newval = val
+        newvals.append(newval)
+    return b"-m %s" % b"".join(newvals)
+
+
+class SGESite(Site):
+    """Site managed using SGE"""
+
+
+    directive_prefix = b"#$ "
+    directive_translate = {
+        "billing_account": b"-A %s",
+        "error_file": b"-e %s",
+        "export_vars": _translate_export_vars,
+        "mail_type": _translate_mail_type,
+        "mail_user": b"-M %s",
+        "name": b"-N %s",
+        "output_file": b"-o %s",
+        "priority": b"-p %s",
+        "queue": b"-q %s",
+        "walltime": b"-l walltime=%s",
```

Comment:
From what I see in the docs, the option corresponding to wall time limit is `s_rt` (or `h_rt` for a hard kill straight away)

```suggestion
        "walltime": b"-l s_rt=%s",
```

Reply:
I modify it to h_rt. It's what we normally use and is a bit safer